pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,493 @@
+	oldNode := oldObj.(*corev1.Node)
+	node := newObj.(*corev1.Node)
+	m.latencyStore.updateNode(oldNode, node)
+
+	klog.InfoS("Node updated", "Node", klog.KObj(node))
+}
+
+// onNodeDelete is the event handler for deleting Node.
+func (m *NodeLatencyMonitor) onNodeDelete(obj interface{}) {
+	// Check if the object is a not a node
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		return
+	}
+
+	m.latencyStore.deleteNode(node)
+}
+
+// onNodeLatencyMonitorAdd is the event handler for adding NodeLatencyMonitor.
+func (m *NodeLatencyMonitor) onNodeLatencyMonitorAdd(obj interface{}) {
+	nlm := obj.(*v1alpha1.NodeLatencyMonitor)
+	klog.InfoS("NodeLatencyMonitor added", "NodeLatencyMonitor", klog.KObj(nlm))
+
+	if err := m.updateLatencyConfig(nlm); err != nil {
+		klog.ErrorS(err, "Failed to update latency config")
+	}
+}
+
+// onNodeLatencyMonitorUpdate is the event handler for updating NodeLatencyMonitor.
+func (m *NodeLatencyMonitor) onNodeLatencyMonitorUpdate(oldObj, newObj interface{}) {
+	oldNLM := oldObj.(*v1alpha1.NodeLatencyMonitor)
+	newNLM := newObj.(*v1alpha1.NodeLatencyMonitor)
+	klog.InfoS("NodeLatencyMonitor updated", "NodeLatencyMonitor", klog.KObj(newNLM))
+
+	if oldNLM.GetGeneration() == newNLM.GetGeneration() {
+		return
+	}
+
+	if err := m.updateLatencyConfig(newNLM); err != nil {
+		klog.ErrorS(err, "Failed to update latency config")
+	}
+}
+
+func (m *NodeLatencyMonitor) updateLatencyConfig(nlm *v1alpha1.NodeLatencyMonitor) error {
+	// Parse the ping interval
+	pingInterval := time.Duration(nlm.Spec.PingIntervalSeconds) * time.Second
+
+	// Update the latency config
+	m.latencyConfig = &LatencyConfig{
+		Enable:   true,
+		Interval: pingInterval,
+	}
+
+	// Notify the latency config changed
+	m.latencyConfigChanged <- struct{}{}
+
+	return nil
+}
+
+// onNodeLatencyMonitorDelete is the event handler for deleting NodeLatencyMonitor.
+func (m *NodeLatencyMonitor) onNodeLatencyMonitorDelete(obj interface{}) {
+	// Update the latency config
+	m.latencyConfig = &LatencyConfig{Enable: false}
+
+	// Notify the latency config changed
+	m.latencyConfigChanged <- struct{}{}
+}
+
+func (m *NodeLatencyMonitor) sendPing(socket net.PacketConn, addr net.IP) error {
+	var requestType icmp.Type
+
+	// Resolve the IP address
+	ip := &net.IPAddr{IP: addr}
+
+	// Create a new ICMP packet
+	if addr.To4() == nil {
+		requestType = ipv6.ICMPTypeEchoRequest
+	} else {
+		requestType = ipv4.ICMPTypeEcho
+	}
+
+	timeStart := time.Now()
+	seqID := getICMPSeq()
+	body := &icmp.Echo{
+		ID:   os.Getpid() & 0xffff,                       // Use the current process ID as the ICMP ID
+		Seq:  int(seqID),                                 // Use the current sequence number as the ICMP sequence number
+		Data: []byte(timeStart.Format(time.RFC3339Nano)), // Store the current time in the ICMP data
+	}
+	msg := icmp.Message{
+		Type: requestType,
+		Code: 0,
+		Body: body,
+	}
+	klog.InfoS("Send ICMP message", "IP", ip, "SeqID", seqID, "body", body)
```

Comment:
same, much too verbose for something that happens super frequently. Maybe it should be `V(4)` as well (or `V(5)`?

Reply:
Yes, it‘s just for debug, I will set the log level to `V(4)`.